gravitee-am-ui/src/app/domain/settings/theme/theme.component.ts

```diff
@@ -0,0 +1,127 @@
+    { 'name': 'deep-red', 'primaryButtonColorHex': '#C52852', 'primaryTextColorHex': '#FFFFFF', 'secondaryButtonColorHex': '#F1C9D4' },
+    { 'name': 'deep-orange', 'primaryButtonColorHex': '#D94C3E', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#F6D2CF' },
+    { 'name': 'orange', 'primaryButtonColorHex': '#EA9345', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#FAE4D0' },
+    { 'name': 'yellow', 'primaryButtonColorHex': '#EED052', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#FBF3D4' },
+    { 'name': 'light-green', 'primaryButtonColorHex': '#9FC929', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#E7F2C9' },
+    { 'name': 'green', 'primaryButtonColorHex': '#7FCF79', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#DFF3DE' },
+    { 'name': 'teal', 'primaryButtonColorHex': '#66C2AC', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#D9F0EA' },
+    { 'name': 'grey', 'primaryButtonColorHex': '#8A979E', 'primaryTextColorHex': '#000000', 'secondaryButtonColorHex': '#E2E5E7' },
+    { 'name': 'deep-grey', 'primaryButtonColorHex': '#252829', 'primaryTextColorHex': '#FFFFFF', 'secondaryButtonColorHex': '#C8C9C9' }
+  ];
+  selectedColorPalette: string;
+  config: any = { lineNumbers: true };
+  formChanged: boolean;
+
+  constructor(private snackbarService: SnackbarService,
+              private router: Router,
+              private route: ActivatedRoute,
+              private authService: AuthService,
+              private themeService: ThemeService,
+              private dialogService: DialogService) {
+  }
+
+  ngOnInit() {
+    this.envId = this.route.snapshot.params['envHrid'];
+    this.domain = this.route.snapshot.data['domain'];
+    this.themes =  this.route.snapshot.data['themes'] || [];
+    this.theme = this.themes[0] || {};
+    if (this.theme.id && this.theme.primaryButtonColorHex) {
+      const filteredObj = _.find(this.colorPalettes, { 'primaryButtonColorHex' : this.theme.primaryButtonColorHex });
+      if (filteredObj) {
+        this.selectedColorPalette = filteredObj.name;
+      }
+    }
+    if (this.theme.logoWidth != undefined && this.theme.logoWidth === 0) {
+      this.theme.logoWidth = null;
+    }
+    this.readonly = !this.authService.hasPermissions(['domain_theme_update']);
+  }
+
+  onModelChange(event) {
+    this.formChanged = true;
+  }
+
+  reset() {
+    this.dialogService
+      .confirm('Reset Theme', 'Are you sure you want to reset the theme?')
+      .subscribe(res => {
+        if (res) {
+          if (this.theme.id) {
+            this.themeService.delete(this.domain.id, this.theme.id).subscribe(() => {
+              this.snackbarService.open('Theme reset');
+              this.themes = [];
+              this.theme = {};
+              this.selectedColorPalette = null;
+            });
+          }
+        }
+      });
+  }
+
+  publish() {
+    let themeToPublish: any = {};
```

Comment:
should be constant, helps in future as the next developer knows the variable is not overwritten.

Reply:
FP